### PR TITLE
send count of Golang clients to PhoneHome app

### PR DESCRIPTION
**Pull request by hasancelik**
this PR is backport of https://github.com/hazelcast/hazelcast/pull/12150 as well

callhome repo counterpart --> https://github.com/hazelcast/callhome/pull/8

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java [hunk 1]
```diff
@@ -166,6 +166,7 @@ public String convertToLetter(int size) {
         parameterCreator.addParam("cjv", Integer.toString(clusterClientStats.get(ClientType.JAVA)));
         parameterCreator.addParam("cnjs", Integer.toString(clusterClientStats.get(ClientType.NODEJS)));
         parameterCreator.addParam("cpy", Integer.toString(clusterClientStats.get(ClientType.PYTHON)));
+        parameterCreator.addParam("cgo", Integer.toString(clusterClientStats.get(ClientType.GO)));
```
Comment (mustafaiman) [inline hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java]: This should also be in master branch, right?

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java [hunk 2]
```diff
@@ -596,6 +597,9 @@ public Operation getPreJoinOperation() {
                 case PYTHON:
                     numberOfPythonClients++;
                     break;
+                case GO:
```
Comment (mustafaiman) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java]: This should also be in master branch, right?

File: hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java (hunk 1 above)
Comment (hasancelik) [reply hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java]: Yes, I will forward-port these cahnges to master 👍

Comment (hasancelik) [reply hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java]: @mustafaiman here is the forward port PR --> https://github.com/hazelcast/hazelcast/pull/12719

**Pull request merged**